repository/comment/repo_comment.go

```diff
@@ -0,0 +1,104 @@
+package comment
+
+import (
+	"database/sql"
+	"fmt"
+	"log/slog"
+	"time"
+
+	"github.com/go-park-mail-ru/2023_2_Vkladyshi/configs"
+
+	_ "github.com/jackc/pgx/stdlib"
+)
+
+type ICommentRepo interface {
+	GetFilmRating(filmId uint64) (float64, uint64, error)
+	GetFilmComments(filmId uint64, first uint64, limit uint64) ([]CommentItem, error)
+	AddComment(filmId uint64, userId uint64, rating uint16, text string) error
+}
+
+type RepoPostgre struct {
+	DB *sql.DB
+}
+
+func GetCommentRepo(config configs.DbDsnCfg, lg *slog.Logger) *RepoPostgre {
+	dsn := fmt.Sprintf("user=%s dbname=%s password= %s host=%s port=%d sslmode=%s",
+		config.User, config.DbName, config.Password, config.Host, config.Port, config.Sslmode)
+	db, err := sql.Open("pgx", dsn)
+	if err != nil {
+		lg.Error("sql open error", "err", err.Error())
+		return nil
+	}
+	err = db.Ping()
+	if err != nil {
+		lg.Error("sql ping error", "err", err.Error())
+		return nil
+	}
+	db.SetMaxOpenConns(config.MaxOpenConns)
+
+	postgreDb := RepoPostgre{DB: db}
+
+	go postgreDb.pingDb(config.Timer, lg)
+	return &postgreDb
+}
+
+func (repo *RepoPostgre) pingDb(timer uint32, lg *slog.Logger) {
+	err := repo.DB.Ping()
+	if err != nil {
+		lg.Error("Repo Comment db ping error", "err", err.Error())
+	}
+
+	time.Sleep(time.Duration(timer) * time.Second)
+}
+
+func (repo *RepoPostgre) GetFilmRating(filmId uint64) (float64, uint64, error) {
+	var rating float64
+	var number uint64
+	err := repo.DB.QueryRow(
+		"SELECT AVG(rating), COUNT(rating) FROM users_comment "+
+			"WHERE id_film = $1", filmId).Scan(&rating, &number)
+	if err == sql.ErrNoRows {
+		return 0, 0, nil
+	}
+	if err != nil {
+		return 0, 0, err
+	}
```

Comment:
В golang не стоит сравнивать ошибку через ==. Лучше использовать из стандартного пакета errors.Is. + Лучше сравнивать как-то так - и нагляднее и внутреннюю магию errors.Is позволяет скипнуть, если err == nil

```suggestion
	if err != nil {
		if errors.Is(err, sql.ErrNoRows) {
			return 0, 0, nil
		}

		return 0, 0, err
	}
```